### PR TITLE
[WIP] Fixes #20430 - katello-change-hostname performs installer dry-run

### PR DESCRIPTION
**Pull request by johnpmitsch**
The installer will do a dry run in the precheck to check for
potential problems before making changes.

### CODE REVIEW

File: katello/katello-change-hostname [hunk 1]
```diff
@@ -197,6 +208,15 @@ def precheck(response)
     fail_with_message("Please specify a hostname.", opt_parser)
   end
 
+  installer_precheck = installer_command + " -d"
```
Comment (ehelms) [inline katello/katello-change-hostname]: I think you mean `-n` here?

**Comment by johnpmitsch** [comment]
@ehelms updated

**Comment by johnpmitsch** [comment]
@ehelms which do we want to do?

```
dry run of the installer (exit if it fails)
update hostname on system + remove certs
real run of the installer
```
or
```
update hostname on the system + remove certs
dry run of the installer (exit if it fails)
real run of the installer
```

My concern with the second option is there would be permanent changes. However, with the first option, the dry run of the installer isn't as thorough of a check.

**Comment by johnpmitsch** [comment]
@ehelms 
Apologies for the stream of consciousness, but here is what I'm thinking for the hostname-change pre-check
```
update hostname on system
move certs to files/folders in a directory we create in /tmp
Perform dry run of the installer
if dry run is successful:
  continue with change hostname script, which will run the actual installer
else:
  move certs back to appropriate location
  change system hostname back to the old hostname
  fail out of script w/ error
```
I think this will be the most robust way to check if the installer will succeed when running katello-change hostname while preventing irreversable changes from happening to user's systems. Curious to hear your thoughts

**Comment by ehelms** [comment]
@johnpmitsch planning to revisit this?

**Comment by johnpmitsch** [comment]
I am closing this as the dry run of the installer would fail sometimes and the actual run would pass. I am working on another "cleanup" solution.

**Pull request closed without merging**